tests/run/fused_types.pyx

```diff
@@ -487,7 +487,8 @@ def test_fused_in_check():
 
 ### see GH3642 - presence of cdef inside "unrelated" caused a type to be incorrectly inferred
 cdef unrelated(cython.floating x):
```

Comment:
Sorry, I was referring to this function, which is unused. Or does it have to be unused?

Reply:
I don't think it particularly matters if it's used or not.

I think the bug is actually surprisingly general and happens all over the place, but most of the time is hidden because floats/doubles are convertible (but here it shows up because the pointers aren't).

I've added a cast to void for the function to try to get rid of the warning, but I think it should still be a valid test if the function is called somewhere instead.

Reply:
`unrelated` probably isn't needed at all. The bug's probably triggered by https://github.com/cython/cython/blob/f808448f2a4493965703261ab5ae2c4743c118e8/tests/run/fused_types.pyx#L287

But relying on that in the test is probably a bit of an obscure side effect.